Doc/whatsnew/3.14.rst

```diff
@@ -668,6 +668,25 @@ io
   file's bytes in full. (Contributed by Cody Maloney and Victor Stinner in
   :gh:`120754` and :gh:`90102`.)
 
+
+uuid
+----
+
+* Improve generation of :class:`~uuid.UUID` objects via their dedicated
+  functions:
+
+  * :func:`~uuid.uuid3` is 47% faster for 16-byte names and 13% faster
+    for 1024-byte names. Performances for longer names remain unchanged.
+  * :func:`~uuid.uuid5` is 35% faster for 16-byte names and 24% faster
+    for 1024-byte names. Performances for longer names remain unchanged.
+  * :func:`~uuid.uuid4` is 33% faster and :func:`~uuid.uuid8` is 38% faster.
+
+  Overall, dedicated generation of UUID objects version 3, 4, 5, and 8 is
```

Comment:
Since you already have the specific improvements I would remove this paragraph (or the other way around)

Reply:
I think it's nice for people who just want an overview (like a TL;DR)

Reply:
Ok, I've removed it since I kept it in the simple changelog.